tests/test_distributions.py

```diff
@@ -123,6 +157,21 @@ def test_check_distribution_compatibility() -> None:
     )
 
     # test dynamic value range (except CategoricalDistribution)
+    distributions.check_distribution_compatibility(
+        EXAMPLE_DISTRIBUTIONS["i"], distributions.IntDistribution(low=-3, high=2)
+    )
+    distributions.check_distribution_compatibility(
+        EXAMPLE_DISTRIBUTIONS["il"], distributions.IntDistribution(low=1, high=13, log=True)
+    )
+    distributions.check_distribution_compatibility(
+        EXAMPLE_DISTRIBUTIONS["f"], distributions.FloatDistribution(low=-3.0, high=-2.0)
+    )
+    distributions.check_distribution_compatibility(
+        EXAMPLE_DISTRIBUTIONS["fl"], distributions.FloatDistribution(low=0.1, high=1.0, log=True)
+    )
+    distributions.check_distribution_compatibility(
+        EXAMPLE_DISTRIBUTIONS["fd"], distributions.FloatDistribution(low=-1.0, high=11.0, step=0.5)
+    )
```

Comment:
In the current implementation, i, il and id are compatible each other. To explicitly test it, you can check them here. Also, it can be applied to f, fl and fd.

Reply:
Thank you very much for pointing out, added the tests.